.pre-commit-config.yaml

```diff
@@ -100,6 +100,57 @@ repos:
         pass_filenames: false
         language: system
 
+      # Javascript linting, formatting.
+      # We're using the local node_modules to simplify things, otherwise would
+      # need an easy way to keep versions in sync. Furthermore some repositories
+      # are either not on pre-commit mirrors or are missing some tagged versions.
+      # pre-commit-managed nodeenv environments also do not work out of the box because
+      # additional_dependencies does not install a flattened dependency tree which is needed by,
+      # for example, our eslint setup.
+
+      - id: biome
+        name: biome (javascript, typescript, json)
+        language: system
+        types: [text]
+        files: \.(jsx?|tsx?|css)$
+        entry: ./node_modules/.bin/biome check --log-level=error --write --files-ignore-unknown=true --no-errors-on-unmatched
+
+      - id: eslint
+        name: eslint
+        language: system
+        files: \.[jt]sx?$
+        entry: ./node_modules/.bin/eslint --quiet --fix
+
+      - id: stylelint
+        name: stylelint
+        language: system
+        files: \.[jt]sx?$
+        entry: ./node_modules/.bin/stylelint --quiet
+
+      - id: prettier
+        name: prettier (yaml, markdown, tsx, jsx, css)
+        language: system
+        # TODO: Remove tsx and jsx when Biome supports styled CSS formatting.
+        types_or: [yaml, markdown, tsx, jsx, css]
+        entry: ./node_modules/.bin/prettier --log-level=error --write
+        # https://pre-commit.com/#regular-expressions
+        exclude: |
+          (?x)^($^
+            # prettier crashes on some of these, by intent:
```

Comment:
crashes or has errors?

Reply:
I have no idea, this was from the previous hook.